share/rocm/cmake/ROCMCreatePackage.cmake

```diff
@@ -218,6 +218,30 @@ macro(rocm_set_cpack_gen)
     endif()
 endmacro()
 
+macro(rocm_set_cpack_compression)
+    # Default to xz unless otherwise specified.
+    # The CPack default compression type is gzip, but xz packages are much smaller.
+    # Native packages have defaulted to xz since at least Ubuntu 14.04.
+    if(NOT CPACK_DEBIAN_COMPRESSION_TYPE)
+        if(DEFINED ENV{ROCM_DEBIAN_COMPRESSION_TYPE})
+            set(CPACK_DEBIAN_COMPRESSION_TYPE $ENV{ROCM_DEBIAN_COMPRESSION_TYPE})
+        else()
+            set(CPACK_DEBIAN_COMPRESSION_TYPE "xz")
+        endif()
+    endif()
+
+    # xz compression is slow, but can be parallelized with CMake 3.20 or later.
+    # The compression ratio is marginally affected. The speedup is subject to
+    # diminishing returns at higher thread counts.
+    if(NOT CPACK_THREADS)
```

Comment:
Rather than doing `if(NOT CPACK_THREADS)` can we just make these cached variables instead?

Reply:
I initially agreed with you, but I took a look at how CMake handles this sort of thing for the `CMAKE_CXX_COMPILER_LAUNCHER` cache and environment variable, since that has a similar behaviour. [They use exactly the same sort of check](https://github.com/Kitware/CMake/blob/v3.23.1/Modules/CMakeCXXInformation.cmake#L219-L222), despite it being seemingly redundant.

I think the reasoning is probably that they don't want the variable to show up as an editable cache entry if the library build code is overriding the cache entry with a normal variable. It would be kind of annoying to edit the cache entry only to discover that it doesn't have any effect.

So, instead I made them cache variables in addition to doing `if(NOT CPACK_THREADS)`.

Reply:
I asked in the cmake channel of the cpplang slack about why CMake did it that way and Alex Reinking pointed me to the bug report that lead to CMP0126: [Set-cache deletes normal variables unpredictably (#22038)](https://gitlab.kitware.com/cmake/cmake/-/issues/22038).